experiment_scheduler/master/process_monitor.py

```diff
@@ -61,34 +63,40 @@ def _get_stubs(self) -> Dict[str, Any]:
             stubs[address] = TaskManagerStub(channel)
         return stubs
 
-    def _health_check(self, thread_queue, time_interval=5):
-        # move to decorator later
+    def _health_check(self, thread_queue, time_interval=1):
+        """
+        Check the status of task manager and task status.
+        If there are finished tasks, then update the status column of Task Table
+        :param thread_queue:
+        :param time_interval:
+        """
         while True:
             for task_manager in self.task_manager_address:
                 try:
-                    self.task_manager_stubs[task_manager].health_check(PROTO_EMPTY)
+                    server_status = self.task_manager_stubs[task_manager].health_check(PROTO_EMPTY)
+                    if server_status.alive:
+                        if self.selected_task_manager == -1:
+                            self.selected_task_manager = 1
+                    if len(server_status.task_id_array) > 0:
+                        for task_id in server_status.task_id_array:
+                            task = TaskEntity.get(id=task_id)
+                            task.status = 2
+                            task.commit()
                     thread_queue[f"is_{task_manager}_healthy"] = True
                 except RpcError as error:
-                    thread_queue[f"is_{task_manager}_healthy"] = False
-                    self.logger.error(
-                        "currently task manager %s is not available.\n error log : %s",
-                        task_manager,
-                        error,
-                    )
+                    if self.selected_task_manager != -1:
+                        thread_queue[f"is_{task_manager}_healthy"] = False
+                        self.logger.error(
+                            "currently task manager %s is not available.\n error log : %s",
+                            task_manager,
+                            error,
+                        )
+                    else:
+                        self.logger.warning("task managers are not started yet")
```

Comment:
Process monitor check the task_managers with status and finished tasks.
And we change the logger message when task_managers is not started and when task_manager is abnormal.